integration/network/network_linux_test.go

```diff
@@ -387,9 +388,83 @@ func checkCtrRoutes(t *testing.T, ctx context.Context, apiClient client.APIClien
 		return s == ""
 	})
 
-	assert.Equal(t, len(routes), expRoutes, "expected %d routes, got %d:\n%s", expRoutes, len(routes), strings.Join(routes, "\n"))
+	assert.Check(t, is.Equal(len(routes), expRoutes), "expected %d routes, got %d:\n%s", expRoutes, len(routes), strings.Join(routes, "\n"))
 	defFound := slices.ContainsFunc(routes, func(s string) bool {
 		return strings.Contains(s, expDefRoute)
 	})
-	assert.Assert(t, defFound, "default route %q not found:\n%s", expDefRoute, strings.Join(routes, "\n"))
+	assert.Check(t, defFound, "default route %q not found:\n%s", expDefRoute, strings.Join(routes, "\n"))
+}
+
+// TestMixL3IPVlanAndBridge checks that a container can be connected to a layer-3
+// ipvlan network as well as a bridge ... the bridge network will set up a
+// default gateway, if selected as the gateway endpoint. The ipvlan driver sets
+// up a connected route to 0.0.0.0 or [::], a route via a specific interface with
+// no next-hop address (because the next-hop address can't be ARP'd to determine
+// the interface). These two types of route cannot be set up at the same time.
+// So, the ipvlan's route must be treated like the default gateway and only get
+// set up when the ipvlan is selected as the gateway endpoint.
+// Regression test for https://github.com/moby/moby/issues/48576
+func TestMixL3IPVlanAndBridge(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows", "no ipvlan on Windows")
```

Comment:
Perhaps (not for this PR) we should add a utility to check whether the daemon supports `network-driver X`; `docker info` should return this, so we could use that for detecting instead of based on platform;

```bash
docker info --format='{{ json .Plugins.Network }}'
["bridge","host","ipvlan","macvlan","null","overlay"]
```